lib/spack/spack/cmd/stack.py

```
@@ -0,0 +1,120 @@
+since the original files need to remain in place because dependencies are found
+via RPATH (which are the old paths in the remote repository).
+"""
+
+from __future__ import print_function
+
+__author__ = "Oliver Breitwieser (UHEI)"
+
+import os
+import os.path as osp
+import spack
+import spack.config
+import spack.directory_layout
+import spack.store
+from spack.filesystem_view import filter_exclude
+from spack.util.path import canonicalize_path
+import llnl.util.tty as tty
+import llnl.util.filesystem as fs
+
+description = "integrate packages of remote spack repositories into this one"
+section = "admin"
+level = "long"  # TODO: re-check what 'level' is supposed to mean
+
+
+def setup_parser(sp):
+    setup_parser.parser = sp
+
+    sp.add_argument(
+        '-v', '--verbose', action='store_true', default=False,
+        help="If not verbose only warnings/errors will be printed.")
+    sp.add_argument(
+        '-e', '--exclude', action='append', default=[],
+        help="exclude packages with names matching the given regex pattern")
+
+    sp.add_argument("remotes", nargs="+",
+                    metavar='remote', action='store',
+                    help="one or more remote spack repositories which "
+                         "contain packages to be added as external packages")
+
+
+def add_remote_packages(remote, exclude=[], verbose=False):
+    """
+        Add all installed packages in `remote` to the packages dictionary.
+    """
+    config = spack.config.get_config("config")
+
+    # NOTE: This has to be kept in sync with spack/store.py!
+    layout = spack.directory_layout.YamlDirectoryLayout(
+        canonicalize_path(osp.join(remote, 'opt', 'spack')),
+        hash_len=config.get('install_hash_length'),
+        path_scheme=config.get('install_path_scheme'))
+
+    num_packages = 0
+
+    for spec in filter_exclude(layout.all_specs(), exclude):
+        src = layout.path_for_spec(spec)
```

this will probably fail if the remote spack config is different from the local spack config. Not sure if it's worth the bother though... Supporting differently configured spack's seems to be inviting trouble, but if the configs aren't identical we should probably throw a comprehensible error message

---

That would require adding support to `config.py` to support different spack-repositories - be my guest.. :o)

I find it infeasible from a effort/usefulness standpoint.

---

Ack, my comment is mostly: Let's be nice to users